src/tests/ftest/control/daos_snapshot.py

```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/python
+'''
+  (C) Copyright 2020 Intel Corporation.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+  The Government's rights to use, modify, reproduce, release, perform, display,
+  or disclose this software are subject to the terms of the Apache License as
+  provided in Contract No. B609815.
+  Any reproduction of computer software, computer software documentation, or
+  portions thereof marked with this legend must also reproduce the markings.
+'''
+from apricot import TestWithServers, skipForTicket
+from daos_utils import DaosCommand
+from control_test_base import ControlTestBase
+
+
+class DaosSnapshotTest(ControlTestBase):
+    """
+    Test Class Description:
+        Test daos container create-snap, list-snaps, and destroy-snap. Test
+        steps:
+        1. Create 5 snapshots. Obtain the epoch value each time we create a
+            snapshot.
+        2. List the 5 snapshots, obtain the epoch values, and compare against
+            those returned during create-snap.
+        3. Destroy all the snapshots one by one.
+        4. List and verify that there's no snapshot.
+
+        Test destroy-snap --epcrange.
+        1. Create 5 snapshots.
+        2. Destory all the snapshots with --epcrange. Use the first epoch for B
+            and the last epoch for E.
+        3. List and verify that there's no snapshot.
+
+    Note that we keep the test steps basic due to time constraint. Add more
+    cases if we see bugs around this feature.
+
+    :avocado: recursive
+    """
+    def __init__(self, *args, **kwargs):
+        """Initialize a DaosSnapshotTest object."""
+        super(DaosSnapshotTest, self).__init__(*args, **kwargs)
+        self.daos_cmd = None
+
+    def create_snapshot(self, pool_uuid, svc, cont_uuid, count):
+        """Create snapshots and return the epoch values obtained from stdout.
+
+        Args:
+            pool_uuid (String): Pool UUID.
+            svc (String): Service replicas.
+            cont_uuid (String): Container UUID.
+            count (Integer): Number of snapshots to create.
+
+        Returns:
+            List of string: Epochs obtained from stdout.
+        """
+        kwargs = {"pool": pool_uuid, "svc": svc, "cont": cont_uuid}
+        epochs = []
+        for _ in range(count):
+            epochs.append(
+                self.daos_cmd.get_output("container_create_snap", **kwargs)[0])
```

Comment:
The usage of `get_output()` should be deprecated in favor of calling the command method, in this case `container_create_snap(**kwargs)`, directly.  Ideally we should also be updating these command methods to handle the regex of the command output internally to make them easier to use.

Reply:
Done. Thanks for reminding. I remember doing this for dmg storage scan.